core/tests/net/tcp.cpp

```diff
@@ -0,0 +1,98 @@
+#include <cstring>
+#include <thread>
+
+#include <doctest/doctest.h>
+
+#include <cubos/core/net/address.hpp>
+#include <cubos/core/net/tcp_listener.hpp>
+#include <cubos/core/net/tcp_stream.hpp>
+
+TEST_CASE("tcp")
+{
+    using cubos::core::net::Address;
+    using cubos::core::net::TcpListener;
+    using cubos::core::net::TcpStream;
+
+    SUBCASE("write and read with stream and listener")
+    {
+        auto svAddr = Address::fromIPv4(127, 0, 0, 1);
+        uint16_t svPort = 8080;
+
+        std::thread serverThread{[&svAddr, &svPort]() {
+            TcpListener listener;
+            REQUIRE(listener.listen(svAddr, svPort, 1));
+
+            TcpStream stream;
+            REQUIRE(listener.accept(stream));
+
+            char readBuffer[4 + 1];
+            REQUIRE(stream.read(readBuffer, sizeof(readBuffer)) > 0);
+            readBuffer[4] = '\0';
+            REQUIRE(std::string(readBuffer) == "ping");
+
+            REQUIRE(stream.write("pong", 4) > 0);
```

Comment:
Same as above, I think you also need to repeat this call until everything has been written.